routing/datasource.go

```diff
@@ -528,6 +531,69 @@ type routeTable struct {
 	created       time.Time
 }
 
+func fadeInState(now time.Time, duration time.Duration, detected time.Time) (time.Duration, bool) {
+	rel := now.Sub(detected)
+	return rel, rel > 0 && rel < duration
+}
+
+func fadeIn(now time.Time, duration time.Duration, exponent float64, detected time.Time) float64 {
+	rel, fadingIn := fadeInState(now, duration, detected)
+	if !fadingIn {
+		return 1
+	}
+
+	return math.Pow(float64(rel)/float64(duration), exponent)
+}
+
+func goodEndpoint(rnd *rand.Rand, ep *LBEndpoint, route *Route) bool {
+	now := time.Now()
+	f := fadeIn(
+		now,
+		route.LBFadeInDuration,
+		route.LBFadeInExponent,
+		ep.Detected,
+	)
+	goodFadeIn := f == 1 || rnd.Float64() < f
+
+	return goodFadeIn
+}
```

Comment:
This should also work for routes that do not use fadeIn.
Note that pre-existing code contains check like this:
```go
if ctx.Route.LBFadeInDuration <= 0 {
  return ctx.Route.LBEndpoints[i]
}
```

Reply:
`fadeIn` returns 1 when endpoint already passed "fading-in" state.